lib/repl.js

```diff
@@ -494,14 +494,17 @@ function REPLServer(prompt,
   self.writer = options.writer || exports.writer;
 
   if (options.useColors === undefined) {
-    options.useColors = self.terminal;
+    options.useColors = self.terminal && process.env.TERM !== 'dumb';
```

Comment:
This could use the `writeStream.getColorDepth()` function to test for color support. It detects more terminals that lack color support and adds color support to dumb terminals that still support colors (`env.TERM = 'dumb' && env.COLORTERM = '1'`).

Reply:
I think original problem is because programs use `dumb` to signal the apps preferred behavior, but don't clean any env variables. So it's used completely unrelated with colors.

For example in MacVim, when you run node with `:!` there are variables that may suggest color support:

```
  TERM_PROGRAM: 'iTerm.app',
  TERM: 'dumb',
  LSCOLORS: 'Gxfxcxdxbxegedabagacad',
  COLORS: '16777216',
  COLORFGBG: '7;0',
  COLORTERM: 'truecolor',
```

Reply:
Yes, it seems like I made a mistake there. I'll open a PR soon to fix that.

Reply:
https://github.com/nodejs/node/pull/26264

Reply:
Replaced check by `.getColorDepth()` check similar to the one in the console.